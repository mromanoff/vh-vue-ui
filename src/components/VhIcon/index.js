import VhIcon from './VhIcon';

export default VhIcon;
export { VhIcon };
