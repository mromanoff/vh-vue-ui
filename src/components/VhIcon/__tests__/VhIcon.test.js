import { shallowMount } from '@vue/test-utils';
import VhIcon from '../VhIcon.vue';

describe.skip('VhIcon.vue', () => {
  let wrapper;

  // TEST functional component with passing props: https://vue-test-utils.vuejs.org/api/options.html#context

  beforeEach(() => {
    wrapper = shallowMount(VhIcon, {
      context: {
        props: { size: 'medium' },
      },
    });
  });

  it('renders default VhIcon', () => {
    expect(wrapper.isFunctionalComponent).toBe(true);
    expect(wrapper.is('button')).toBe(true);
    expect(wrapper.classes()).toStrictEqual(['VhButton', 'VhButton--medium']);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhIcon', () => {
    wrapper = shallowMount(VhIcon, {
      propsData: {
        name: 'Test Name',
        color: 'red',
        size: 'large',
      },
    });
    expect(wrapper.props().color).toBe('red');
    expect(wrapper.props().name).toBe('Test Name');
    expect(wrapper.props().size).toBe('large');
    expect(wrapper).toMatchSnapshot();
  });
});
