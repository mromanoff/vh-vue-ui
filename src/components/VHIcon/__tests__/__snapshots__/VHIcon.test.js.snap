// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VHIcon.vue renders modified VHIcon 1`] = `
<svg xmlns="http://www.w3.org/2000/svg" role="presentation" class="Icon Icon--large Icon--red">
  <use xlink:href="/icons-sprite.svg#Test Name"></use>
</svg>
`;

exports[`VhIcon.vue renders default VhIcon 1`] = `
<svg xmlns="http://www.w3.org/2000/svg" role="presentation" class="Icon Icon--medium Icon--default">
  <title>Test Name icon</title>
  <use xlink:href="/icons-sprite.svg#Test Name"></use>
</svg>
`;

exports[`VhIcon.vue renders modified VhIcon 1`] = `
<svg xmlns="http://www.w3.org/2000/svg" role="presentation" class="Icon Icon--large Icon--red">
  <title>Test Name icon</title>
  <use xlink:href="/icons-sprite.svg#Test Name"></use>
</svg>
`;
